src/vcpkg/tools.cpp

```diff
@@ -25,25 +25,57 @@ namespace vcpkg
         std::string sha512;
     };
 
-    static Optional<std::array<int, 3>> parse_version_string(const std::string& version_as_string)
+    // /\d+\.\d+(\.\d+)?/
+    static Optional<std::array<int, 3>> parse_version_string(StringView string_version)
```

Comment:
While I have outstanding changes to tools.cpp, this change will not create issues for me.